### PR TITLE
Add support to provide api_version

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/12226

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py [hunk 1]
```diff
@@ -67,6 +71,7 @@ def __init__(self, endpoint, credential, **kwargs):
 
         authentication_policy = get_authentication_policy(credential)
         polling_interval = kwargs.pop("polling_interval", POLLING_INTERVAL)
+        validate_api_version(api_version)
```
Comment (rakshith91) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:74]:
```suggestion
        api_version = kwargs.pop('api_version', None)
        validate_api_version(api_version)
```

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_api_versions.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+_SUPPORTED_API_VERSIONS = [
+    "2.0-preview.1",
```
Comment (rakshith91) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_api_versions.py:7]:
```suggestion
    "2.0",
```

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py [hunk 3]
```diff
@@ -42,6 +43,9 @@ class FormRecognizerClient(object):
     :param credential: Credentials needed for the client to connect to Azure.
         This is an instance of AzureKeyCredential if using an API key or a token
         credential from :mod:`azure.identity`.
+    :keyword str api_version:
+        The API version of the service to use for requests.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:47]: Can we add that it defaults to the latest service version?

Comment (rakshith91) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py:47]:
```suggestion
        The API version of the service to use for requests. It defaults to the latest service version.
```

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_api_versions.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_api_versions.py:2]: can we add at least one test for this?

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py [hunk 5]
```diff
@@ -54,6 +55,9 @@ class FormTrainingClient(object):
     :param credential: Credentials needed for the client to connect to Azure.
         This is an instance of AzureKeyCredential if using an API key or a token
         credential from :mod:`azure.identity`.
+    :keyword str api_version:
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py:58]: the api_version docstrings should be moved below the :type credential:

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py [hunk 6]
```diff
@@ -56,6 +57,9 @@ class FormTrainingClient(object):
         credential from :mod:`azure.identity`.
     :type credential: :class:`~azure.core.credentials.AzureKeyCredential` or
         :class:`~azure.core.credentials.TokenCredential`
+    :keyword str api_version:
+        The API version of the service to use for requests.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py:61]:
```suggestion
        The API version of the service to use for requests. It defaults to the latest service version.
```

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/aio/_form_training_client_async.py [hunk 7]
```diff
@@ -57,6 +58,9 @@ class FormTrainingClient(object):
         credential from :mod:`azure.identity`.
     :type credential: :class:`~azure.core.credentials.AzureKeyCredential`
         or :class:`~azure.core.credentials_async.AsyncTokenCredential`
+    :keyword str api_version:
+        The API version of the service to use for requests.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/aio/_form_training_client_async.py:62]:
```suggestion
        The API version of the service to use for requests. It defaults to the latest service version.
```

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py (hunk 6 above)
Comment (rakshith91) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py:61]: wonder how this got changed back -

File: sdk/formrecognizer/azure-ai-formrecognizer/tests/test_mgmt.py [hunk 8]
```diff
@@ -160,3 +160,11 @@ def test_get_form_recognizer_client(self, resource_group, location, form_recogni
                 frc.begin_recognize_receipts_from_url(self.receipt_url_jpg).wait()
             ftc.get_account_properties()
             assert transport.session is not None
+
+    @GlobalFormRecognizerAccountPreparer()
+    def test_api_version_form_recognizer_client(self, resource_group, location, form_recognizer_account, form_recognizer_account_key):
```
Comment (rakshith91) [inline sdk/formrecognizer/azure-ai-formrecognizer/tests/test_mgmt.py:165]:
```suggestion
    def test_api_version_form_training_client(self, resource_group, location, form_recognizer_account, form_recognizer_account_key):
```

**Pull request merged**